server/services/tables.js

```diff
@@ -69,12 +96,15 @@ const addTable = async (table, userCreated) => {
     return 0; //everything good
 }
 
-const leaveTable = async (tableName,username) => {
+const leaveTable = async (tableName, nickname) => {
     try {
-        const table = await tableSchema.findOne({ "name": tableName });
-        const user = await userSchema.findOne({ "username": username });
-        table.playersOnTable.remove(user);
-        await table.save();
+      
+        // Update the table by pulling the player with the given nickname from the array
+        await tableSchema.updateOne(
+          { name: tableName },
+          { $pull: { playersOnTable: { nickname: nickname } } }
+        );
```

Comment:
from the chat. just gets the table with that name in the schema and delete where the nickname in what we wrote.
I got to a situtation when 1 user got in multiple times (fixed it) and it removed all his enteries so its great.